### PR TITLE
Make timesequence_failed test more robust.

### PR DESCRIPTION
**Pull request by jwpeterson**
- Add TimeSequenceStepperFailTest, a special TimeStepper that fails a
  pre-determined step, to simulate failures in actual code.
- Remove bad tolerances from original input file.
- Re-gold and re-enable test on Intel.

Refs #6353.

### CODE REVIEW

File: test/src/timesteppers/TimeSequenceStepperFailTest.C [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  // The goal is to fail exactly one timestep which matches its
+  // original sequence point order, other than the initial condition.
+  // This can only happen once, since after you fail, you are no
+  // longer on the original time sequence (off by one or more).  Since
+  // this can only happen once, we only check the [1]th sequence
+  // point.  (Having fewer than two sequence points doesn't really
+  // make sense...)
+  mooseAssert(_original_time_sequence.size() > 1, "Must have at least two sequence points!");
+
+  if (_t_step == 1 && std::abs(_time - _original_time_sequence[1]) < TOLERANCE)
```
Comment (snschune) [inline test/src/timesteppers/TimeSequenceStepperFailTest.C]: maybe use the `absoluteFuzzyEqual`?

**Comment by snschune** [comment]
👍

**Comment by jwpeterson** [comment]
I added the call to `absoluteFuzzyEqual()`, should be good to go.

**Comment by permcody** [comment]
👍

**Pull request merged**